tests/unit/test_tasks.py

```diff
@@ -225,7 +226,18 @@ def test_cleanups_only_ran_once_on_exception(self):
         for done_callback in done_callbacks:
             self.transfer_coordinator.add_done_callback(done_callback)
 
-        submission_task()
+        try:
+            submission_task()
+        except InvalidStateTransitionError:
```

Comment:
It will not be propagated to "real" code because submission tasks are ran in executors, where if they see an error, they will just set the resulting future's result to the invalid state error. The only chance a user can run into one of these is if they call ``TransferFuture.cancel()`` themselves as there is no other direct hooks to updating the status of the ``TransferCoordintor``.